src/coinbase/actions/sendUserOperation.ts

```diff
@@ -0,0 +1,114 @@
+import { UserOperationCalls } from "viem/_types/account-abstraction";
+import { SmartWallet } from "../wallets/types";
+import { NetworkIdentifier, UserOperationStatusEnum } from "../../client";
+import { encodeFunctionData } from "viem";
+import { Coinbase } from "../coinbase";
+import { wait } from "../utils/wait";
+
+export type SendUserOperationOptions<T extends readonly unknown[]> = {
+  calls: UserOperationCalls<T>;
+};
+
+export type SendUserOperationReturnType = {
+  id: string;
+  networkId: NetworkIdentifier;
+  smartWalletAddress: string;
+  status: UserOperationStatusEnum;
+  wait: () => Promise<SendUserOperationReturnType>;
+};
+
+export async function sendUserOperation<T extends readonly unknown[]>(
+  wallet: SmartWallet,
+  options: SendUserOperationOptions<T>,
+): Promise<SendUserOperationReturnType> {
+  if (!wallet.networkId) {
+    throw new Error("Network not set - call use({networkId}) first");
+  }
+  const networkId = wallet.networkId;
```

Comment:
The following is the conventional way of picking fields off on object in JS
```suggestion
  const { networkId } = wallet;
```

Reply:
FYI this is known as [object destructuring](https://www.w3schools.com/js/js_es6.asp#mark_object_destructuring)

Reply:
nice! changing to this